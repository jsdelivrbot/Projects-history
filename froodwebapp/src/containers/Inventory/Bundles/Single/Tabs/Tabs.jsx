import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-flexbox-grid';
import { Tabs } from 'antd';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import {
  bundleInfoGetParallelRequest,
  bundleItemsGetRequest,
  bundleAssembliesGetRequest
} from 'redux-base/actions';
import {
  ConnectedMainTab,
  ConnectedItemsTab,
  ConnectedAssemblyTab
} from './index';

const mapDispatchToProps = {
  bundleInfoGetParallelRequest,
  bundleItemsGetRequest,
  bundleAssembliesGetRequest
};

const { TabPane } = Tabs;

export class ItemBundlesTabs extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeTab: 'main'
    };
  }

  componentWillMount() {
    this.props.bundleInfoGetParallelRequest({
      params: [{ id: this.props.bundleId }]
    });
  }

  handleTabClick = (activeTab) => {
    const { bundleId } = this.props;
    this.setState({
      activeTab
    }, () => {
      switch (activeTab) {
        case 'main':
          this.props.bundleInfoGetParallelRequest({
            params: [{ id: this.props.bundleId }]
          });
          break;
        case 'items':
          this.props.bundleItemsGetRequest({
            id: bundleId
          });
          break;
        case 'assembly':
          this.props.bundleAssembliesGetRequest({
            id: bundleId
          });
          break;
        default:
      }
    });
  }

  render() {
    const {
      activeTab,
    } = this.state;

    const {
      bundleId
    } = this.props;

    const { isNewBundle } = this.props;

    return (
      <Row>
        <Col lg>
          <Tabs
            activeKey={ activeTab }
            onTabClick={ this.handleTabClick }
            animated={ false }
          >
            <TabPane
              key="main"
              tab="Main"
            >
              <ConnectedMainTab
                isNewBundle={ isNewBundle }
              />
            </TabPane>
            <TabPane
              key="items"
              tab="Items"
            >
              <ConnectedItemsTab
                bundleId={ bundleId }
              />
            </TabPane>
            <TabPane
              key="assembly"
              tab="Assembly"
            >
              <ConnectedAssemblyTab
                bundleId={ bundleId }
              />
            </TabPane>
          </Tabs>
        </Col>
      </Row>
    );
  }
}

ItemBundlesTabs.propTypes = {
  // props
  bundleId: PropTypes.string.isRequired,
  isNewBundle: PropTypes.bool.isRequired,
  // redux-base
  bundleInfoGetParallelRequest: PropTypes.func.isRequired,
  bundleItemsGetRequest: PropTypes.func.isRequired,
  bundleAssembliesGetRequest: PropTypes.func.isRequired,
};

export default withRouter(connect(null, mapDispatchToProps)(ItemBundlesTabs));
