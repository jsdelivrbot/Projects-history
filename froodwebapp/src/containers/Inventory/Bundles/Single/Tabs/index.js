export ConnectedMainTab from './MainTab/MainTab';
export ConnectedItemsTab from './ItemsTab/ItemsTab';
export ConnectedAssemblyTab from './AssemblyTab/AssemblyTab';
export Tabs from './Tabs';
