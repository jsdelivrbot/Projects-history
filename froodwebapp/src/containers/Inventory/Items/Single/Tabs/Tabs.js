import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-flexbox-grid';
import { Tabs } from 'antd';
import { connect } from 'react-redux';
import {
  itemInfoGetParallelRequest,
  itemBinsGetRequest,
  itemSuppliersGetRequest,
  itemStockLifeCycleGetRequest,
  itemUOMGetRequest,
  itemImagesGetRequest
} from 'redux-base/actions';
import { ActionButton } from 'components';
import {
  ConnectedMainTab,
  ConnectedBinsTab,
  ConnectedSuppliersTab,
  StockLifeCycleTab,
  ConnectedUOMTab,
  ImagesTab
} from './index';

const { TabPane } = Tabs;

const mapStateToProps = state => ({
  loadingPage: state.item.loadingPage,
  initialValues: state.item.itemInfo
});

const mapDispatchToProps = {
  itemInfoGetParallelRequest,
  itemBinsGetRequest,
  itemSuppliersGetRequest,
  itemStockLifeCycleGetRequest,
  itemUOMGetRequest,
  itemImagesGetRequest
};


export class ItemsTabs extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeTab: 'main',
      hasInitialProductVariants: false,
      productKeyColumnDisabled: false
    };
  }

  componentWillMount() {
    this.props.itemInfoGetParallelRequest({
      params: [{ id: this.props.itemId }]
    });
  }

  componentWillReceiveProps(nextProps) {
    if (!this.props.isNewSkuItem
       && nextProps.initialValues
       && nextProps.initialValues.options
       && nextProps.initialValues.options.length !== 0) {
      this.setState({
        hasInitialProductVariants: true,
        productKeyColumnDisabled: true
      });
    }
  }

  handleAddProductVariants = () => {
    this.setState({
      hasInitialProductVariants: true
    });
  }

  handleTabClick = (activeTab) => {
    const id = this.props.itemId;
    this.setState({
      activeTab
    }, () => {
      switch (activeTab) {
        case 'main':
          this.props.itemInfoGetParallelRequest({ params: [{ id }] });
          break;
        case 'bins':
          this.props.itemBinsGetRequest({ id });
          break;
        case 'suppliers':
          this.props.itemSuppliersGetRequest({ id });
          break;
        case 'stockLifeCycle':
          this.props.itemStockLifeCycleGetRequest({ id });
          break;
        case 'uom':
          this.props.itemUOMGetRequest({ id });
          break;
        case 'images':
          this.props.itemImagesGetRequest({ id: this.props.itemId });
          break;
        default:
      }
    });
  }

  render() {
    const {
      activeTab,
      hasInitialProductVariants,
      productKeyColumnDisabled
    } = this.state;

    const {
      itemId,
      isNewSkuItem,
      initialValues,
      loadingPage
    } = this.props;
    const hasVariants = initialValues ? initialValues.variants : false;
    return (
      <Row>
        <Col xs sm md lg>
          <Tabs
            activeKey={ activeTab }
            onTabClick={ this.handleTabClick }
            animated={ false }
            tabBarExtraContent={
              !loadingPage
           && activeTab === 'main'
           && !hasInitialProductVariants
           &&
           <ActionButton onClick={ this.handleAddProductVariants }>
              Add Variants
           </ActionButton>
            }
          >
            <TabPane
              key="main"
              tab="Main"
            >
              <ConnectedMainTab
                itemId={ itemId }
                isNewSkuItem={ isNewSkuItem }
                hasInitialProductVariants={ hasInitialProductVariants }
                productKeyColumnDisabled={ productKeyColumnDisabled }
              />
            </TabPane>
            <TabPane
              key="bins"
              tab="Bins"
            >
              <ConnectedBinsTab
                itemId={ itemId }
              />
            </TabPane>
            <TabPane
              key="suppliers"
              tab="Suppliers"
            >
              <ConnectedSuppliersTab
                itemId={ itemId }
              />
            </TabPane>
            <TabPane
              key="uom"
              tab="UOM"
            >
              <ConnectedUOMTab
                itemId={ itemId }
              />
            </TabPane>
            <TabPane
              key="images"
              tab="Images"
            >
              <ImagesTab
                itemId={ itemId }
              />
            </TabPane>
            { hasVariants &&
              <TabPane
                key="stockLifeCycle"
                tab="Stock LifeCycle"
              >
                <StockLifeCycleTab
                  itemId={ itemId }
                />
              </TabPane>
            }
          </Tabs>
        </Col>
      </Row>
    );
  }
}

ItemsTabs.propTypes = {
  // triggers
  loadingPage: PropTypes.bool.isRequired,
  // props
  itemId: PropTypes.string.isRequired,
  isNewSkuItem: PropTypes.bool.isRequired,
  // redux-base
  itemInfoGetParallelRequest: PropTypes.func.isRequired,
  itemBinsGetRequest: PropTypes.func.isRequired,
  itemSuppliersGetRequest: PropTypes.func.isRequired,
  itemStockLifeCycleGetRequest: PropTypes.func.isRequired,
  itemImagesGetRequest: PropTypes.func.isRequired,
  itemUOMGetRequest: PropTypes.func.isRequired,
  // redux-form
  initialValues: PropTypes.object,
};

export default connect(mapStateToProps, mapDispatchToProps)(ItemsTabs);
