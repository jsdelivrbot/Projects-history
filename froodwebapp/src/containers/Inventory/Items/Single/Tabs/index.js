export ConnectedMainTab from './MainTab/MainTab';
export ConnectedBinsTab from './BinsTab/BinsTab';
export ConnectedSuppliersTab from './SuppliersTab/SuppliersTab';
export ConnectedUOMTab from './UOMTab/UOMTab';
export StockLifeCycleTab from './StockLifeCycleTab/StockLifeCycleTab';
export ImagesTab from './ImagesTab/ImagesTab';
export Tabs from './Tabs';
