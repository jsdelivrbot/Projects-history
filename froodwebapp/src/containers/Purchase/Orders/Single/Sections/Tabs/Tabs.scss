.tabs {
  margin-top: 2rem;
}

.newGrnButton {
  margin-left: auto;
}
