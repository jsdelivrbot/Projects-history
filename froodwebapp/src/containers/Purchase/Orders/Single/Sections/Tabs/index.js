export ConnectedOrderTab from './OrderTab/OrderTab';
export ConnectedGRNTab from './GRNTab/GRNTab';
export ConnectedInvoiceTab from './InvoiceTab/InvoiceTab';
