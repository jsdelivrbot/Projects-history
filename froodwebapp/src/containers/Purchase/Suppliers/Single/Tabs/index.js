export ConnectedMainTab from './MainTab/MainTab';
export ConnectedLocationsTab from './LocationsTab/LocationsTab';
export ConnectedNotesTab from './NotesTab/NotesTab';
export ConnectedContactsTab from './ContactsTab/ContactsTab';
export ConnectedOrdersTab from './OrdersTab/OrdersTab';
export ConnectedPriceListTab from './PriceListTab/PriceListTab';
export ConnectedReturnsTab from './ReturnsTab/ReturnsTab';
export Tabs from './Tabs';
