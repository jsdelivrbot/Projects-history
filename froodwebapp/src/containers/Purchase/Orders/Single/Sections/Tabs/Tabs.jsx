import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Tabs } from 'antd';
import { Row, Col } from 'react-flexbox-grid';
import {
  purchaseOrderGetRequest,
  purchaseOrderGRNGetRequest,
  purchaseOrderInvoiceGetRequest
} from 'redux-base/actions';
import { connect } from 'react-redux';
import {
  ConnectedOrderTab,
  ConnectedGRNTab,
  ConnectedInvoiceTab
} from './index';
import { tabs } from './Tabs.scss';

const mapDispatchToProps = {
  purchaseOrderGetRequest,
  purchaseOrderGRNGetRequest,
  purchaseOrderInvoiceGetRequest
};

const { TabPane } = Tabs;

const hideGrnStatuses = ['Draft', 'Authorized', 'Approved', 'Rejected', 'Cancelled', 'Not Saved'];
const showInvoiceStatuses = ['Received', 'Closed'];

class POTabs extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeTab: 'order'
    };
  }

  componentWillReceiveProps(nextProps) {
    // when we are on GRN or Invoice tab and click `New PO`
    // to show Order Tab without request to API
    if (nextProps.status.toLowerCase() === 'not saved') {
      this.setState({
        activeTab: 'order'
      });
    }
  }

  handleTabClick = (activeTab) => {
    this.setState({
      activeTab
    }, () => {
      switch (activeTab) {
        case 'order':
          this.props.purchaseOrderGetRequest({
            params: [{
              id: this.props.orderId
            }]
          });
          break;
        case 'grn':
          this.props.purchaseOrderGRNGetRequest({
            id: this.props.orderId
          });
          break;
        case 'invoice':
          this.props.purchaseOrderInvoiceGetRequest({
            id: this.props.orderId
          });
          break;
        default:
      }
    });
  }

  render() {
    const {
      status,
      orderInfoDisabled,
      handleSubmit
    } = this.props;

    const { activeTab } = this.state;

    const grnTabDisabled = (status === 'Received');

    return (
      <Row className={ tabs }>
        <Col xs>
          <Tabs
            activeKey={ activeTab }
            onTabClick={ this.handleTabClick }
            animated={ false }
          >
            <TabPane
              key="order"
              tab="Order"
            >
              <ConnectedOrderTab
                status={ status }
                orderInfoDisabled={ orderInfoDisabled }
                handleSubmit={ handleSubmit }
              />
            </TabPane>
            { !hideGrnStatuses.includes(status) &&
              <TabPane
                key="grn"
                tab="GRN"
              >
                <ConnectedGRNTab
                  status={ status }
                  disabled={ grnTabDisabled }
                  handleSubmit={ handleSubmit }
                />
              </TabPane>
            }
            { showInvoiceStatuses.includes(status) &&
              <TabPane
                key="invoice"
                tab="Invoice"
              >
                <ConnectedInvoiceTab handleSubmit={ handleSubmit } />
              </TabPane>
            }
          </Tabs>
        </Col>
      </Row>
    );
  }
}

POTabs.propTypes = {
  // data
  orderId: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  orderInfoDisabled: PropTypes.bool.isRequired,
  // redux-base
  purchaseOrderGRNGetRequest: PropTypes.func.isRequired,
  purchaseOrderGetRequest: PropTypes.func.isRequired,
  purchaseOrderInvoiceGetRequest: PropTypes.func.isRequired
};

export default connect(null, mapDispatchToProps)(POTabs);
