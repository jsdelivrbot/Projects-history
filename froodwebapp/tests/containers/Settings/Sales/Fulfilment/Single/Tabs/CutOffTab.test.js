import { CutOffTab } from 'containers/Settings/Sales/Fulfilment/Single/Tabs/CutOffTab';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('CutOffTab', () => {
  const props = {
    loadingPage: false,
    readonly: false,
    initialValues: {},
    fulfilmentId: '1',
    extrasUpdateRequest: jest.fn(),
    handleSubmit: jest.fn()
  };

  it('renders the CutOffTab component', () => {
    const component = shallow(<CutOffTab { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of CutOffTab', () => {
    const component = shallow(<CutOffTab { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });

  it('calls extrasUpdateRequest in handleSave with correct param', () => {
    const formValues = 'Form values';
    const component = shallow(<CutOffTab { ...props } />);
    component.instance().handleSave(formValues);

    expect(props.extrasUpdateRequest.mock.calls.length).toBe(1);

    expect(props.extrasUpdateRequest.mock.calls[0][0]).toEqual({
      id: props.fulfilmentId,
      payload: formValues
    });
  });
});
