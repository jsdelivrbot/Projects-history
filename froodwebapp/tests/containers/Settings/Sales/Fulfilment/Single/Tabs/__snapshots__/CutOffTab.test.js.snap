// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CutOffTab capturing Snapshot of CutOffTab 1`] = `
<div>
  <Spin
    prefixCls="ant-spin"
    size="default"
    spinning={false}
    wrapperClassName=""
  >
    <PageHeader
      bigText="Delivery Cutoffs"
    />
    <form>
      <Row>
        <Col
          lg={6}
          md={6}
          xs={12}
        >
          <Card
            className="card"
          >
            <Row
              middle="xs"
            >
              <Col
                lg={3}
                md={3}
                xs={12}
              >
                Same Day delivery
              </Col>
              <Col
                lg={4}
                md={4}
                xs={12}
              >
                <Field
                  checkedText="ON"
                  component={[Function]}
                  disabled={false}
                  name="sameDay.isActive"
                  unCheckedText="OFF"
                />
              </Col>
            </Row>
            <Row
              middle="xs"
            >
              <Col
                lg={2}
                md={2}
                xs={12}
              >
                Cutoff time
              </Col>
              <Col
                lg={6}
                md={6}
                xs={12}
              >
                <Field
                  component={[Function]}
                  disabled={false}
                  name="sameDay.time"
                  timeFormat="HH:mm"
                />
              </Col>
            </Row>
            <Row
              middle="xs"
            >
              <Col
                lg={12}
                md={12}
                xs={12}
              >
                Minimum time between ordering and delivery
              </Col>
            </Row>
            <Row
              middle="xs"
            >
              <Col
                lg={2}
                md={2}
                xs={12}
              >
                In Minutes
              </Col>
              <Col
                lg={6}
                md={6}
                xs={12}
              >
                <Field
                  component={[Function]}
                  disabled={false}
                  name="sameDay.interval"
                />
              </Col>
            </Row>
          </Card>
        </Col>
        <Col
          lg={6}
          md={6}
          xs={12}
        >
          <Card
            className="card"
          >
            <Row
              middle="xs"
            >
              <Col
                lg={3}
                md={3}
                xs={12}
              >
                Next Day delivery
              </Col>
            </Row>
            <Row
              middle="xs"
            >
              <Col
                lg={2}
                md={2}
                xs={12}
              >
                Cutoff time
              </Col>
              <Col
                lg={6}
                md={6}
                xs={12}
              >
                <Field
                  component={[Function]}
                  disabled={false}
                  name="nextDay.time"
                  timeFormat="HH:mm"
                />
              </Col>
            </Row>
            <Row
              middle="xs"
            >
              <Col
                lg={12}
                md={12}
                xs={12}
              >
                Minimum time between ordering and delivery
              </Col>
            </Row>
            <Row
              middle="xs"
            >
              <Col
                lg={2}
                md={2}
                xs={12}
              >
                In Days
              </Col>
              <Col
                lg={6}
                md={6}
                xs={12}
              >
                <Field
                  component={[Function]}
                  disabled={false}
                  name="nextDay.interval"
                />
              </Col>
            </Row>
          </Card>
        </Col>
      </Row>
      <Row>
        <Col
          lg={6}
          md={6}
          xs={12}
        >
          <Card
            className="card"
          >
            <Row
              middle="xs"
            >
              <Col
                lg={8}
                md={8}
                xs={12}
              >
                Lead days to show calendar for delivery acceptance
              </Col>
            </Row>
            <Row
              middle="xs"
            >
              <Col
                lg={2}
                md={2}
                xs={12}
              >
                In Days
              </Col>
              <Col
                lg={6}
                md={6}
                xs={12}
              >
                <Field
                  component={[Function]}
                  disabled={false}
                  name="calLeadDays"
                />
              </Col>
            </Row>
          </Card>
        </Col>
      </Row>
      <Controls
        cancelButtonVisible={false}
        submitButtonVisible={true}
      />
    </form>
  </Spin>
</div>
`;
